### PR TITLE
Clean up type error display

### PR DESCRIPTION
**Pull request by iwahbe**
Previously, an error message looked like this:

```
    warning: Type checking failed:
    warning: Unexpected type at field "skipMetadataApiCheck":
               expected boolean type, got string type
    warning: Type checking is still experimental. If you believe that a warning is incorrect,
    please let us know by creating an issue at https://github.com/pulumi/pulumi-terraform-bridge/issues.
    This will become a hard error in the future.
```

The same error now looks like this:

```
    warning: Type checking failed:

                 Unexpected type at field "skipMetadataApiCheck":
                     expected boolean type, got string type

             Type checking is still experimental. If you believe that a warning is incorrect,
             please let us know by creating an issue at https://github.com/pulumi/pulumi-terraform-bridge/issues.
             This will become a hard error in the future.
```

There is only one `warning:` block, and every item is nested within the warning block. The heart of the error: the type failures take up the "center" of the terminal, separated by space from the rest of the display.

### CODE REVIEW

**Comment by iwahbe** [comment]
This is almost entirely a cosmetic change. The only non-display change is that we will no longer warn `Type checking failures:` when we are emitting a structured error.

File: pkg/tfbridge/provider.go [hunk 1]
```diff
@@ -588,41 +588,74 @@ func (p *Provider) typeCheckConfig(
...
+
+	// If we don't have any type errors, we can just return.
+	if len(typeFailures) == 0 {
+		return nil
+	}
+
+	// warningSpaces is same length as the string that prefixes a [logger.Warn] invocation.
+	//
+	//                    "warning: "
+	const warningSpaces = "         "
```
Comment (guineveresaenger) [inline pkg/tfbridge/provider.go:613]: Is there a reason tabs won't work here and for `fourSpaces`? I know we can't get tabs to be the same exact spacing as `warning: ` but they're more visible when reading this code.

File: pkg/tfbridge/provider.go [hunk 2]
```diff
@@ -588,41 +588,74 @@ func (p *Provider) typeCheckConfig(
...
+	msg.WriteString(prefix(1) + "Type checking is still experimental. If you believe that a warning is incorrect,\n" +
+		prefix(1) + "please let us know by creating an " +
+		"issue at https://github.com/pulumi/pulumi-terraform-bridge/issues.\n" +
+		prefix(1) + "This will become a hard error in the future.",
+	)
+	logger.Warn(msg.String())
+	return nil
+}
+
+func (p *Provider) convertTypeFailures(urn resource.URN, tf []TypeFailure) *pulumirpc.CheckResponse {
```
Comment (guineveresaenger) [inline pkg/tfbridge/provider.go:641]: Could we please not use `tf` as a shorthand here - it's really easy to think "Terraform" when reading it.

File: pkg/tfbridge/provider.go [hunk 3]
```diff
@@ -588,41 +588,74 @@ func (p *Provider) typeCheckConfig(
...
+	// all.
+	if p.pulumiSchemaSpec == nil {
+		logger.Debug("p.pulumiSchemaSpec == nil, skipping type checking config")
+		return nil
+	}
+
+	iv := NewInputValidator(urn, *p.pulumiSchemaSpec, true)
+	typeFailures := iv.ValidateConfig(news)
+	if validateShouldError {
+		return p.convertTypeFailures(urn, typeFailures)
```
Comment (guineveresaenger) [inline pkg/tfbridge/provider.go:602]: thank you for breaking this away 💟

**Review by guineveresaenger — APPROVED** [review]
A few non-blocking nits.

With regards to your comment on behavioral change: If I understand correctly, the `warning` that was emitted on failure before was basically redundant?

File: pkg/tfbridge/provider.go (hunk 2 above)
Comment (iwahbe) [reply pkg/tfbridge/provider.go:641]: Sure.

File: pkg/tfbridge/provider.go (hunk 1 above)
Comment (iwahbe) [reply pkg/tfbridge/provider.go:613]: Tabs didn't display the same way for me in my in-editor (emacs/vterm) terminal and my separate terminal (Alacritty).

**Pull request merged**